File: src/ros2_bridge/mappers/foxglove_msgs/location_fixes.rs
//! Typed mapper for `foxglove_msgs/msg/LocationFixes`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn location_fixes_to_bus(msg: ros_env::foxglove_msgs::msg::LocationFixes) -> crate::foxglove_msgs::msg::v1::LocationFixes {
    crate::foxglove_msgs::msg::v1::LocationFixes {
        fixes: msg.fixes.into_iter().map(crate::ros2_bridge::mappers::foxglove_msgs::location_fix::location_fix_to_bus).collect(),
    }
}

pub(crate) fn location_fixes_to_ros(bus: crate::foxglove_msgs::msg::v1::LocationFixes) -> ros_env::foxglove_msgs::msg::LocationFixes {
    ros_env::foxglove_msgs::msg::LocationFixes {
        fixes: bus.fixes.into_iter().map(crate::ros2_bridge::mappers::foxglove_msgs::location_fix::location_fix_to_ros).collect(),
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct FoxgloveMsgsLocationFixesMapper;

impl TypedTopicMapper for FoxgloveMsgsLocationFixesMapper {
    type Ros = ros_env::foxglove_msgs::msg::LocationFixes;
    type Bus = crate::foxglove_msgs::msg::v1::LocationFixes;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(location_fixes_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(location_fixes_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/foxglove_msgs/location_fixes.py
"""Generated mapper for `foxglove_msgs/msg/LocationFixes`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert
from robot_bus.ros2_bridge.mappers.foxglove_msgs.location_fix import location_fix_to_bus, location_fix_to_ros

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.foxglove_msgs.msg.v1 import LocationFixes as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def location_fixes_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.fixes.extend([location_fix_to_bus(x) for x in msg.fixes])
    return bus


def location_fixes_to_ros(bus):
    from foxglove_msgs.msg import LocationFixes as RosMsg

    out = RosMsg()
    out.fixes = [location_fix_to_ros(x) for x in bus.fixes]
    return out


class FoxgloveMsgsLocationFixesMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from foxglove_msgs.msg import LocationFixes as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return location_fixes_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return location_fixes_to_ros(bus)
